### ISSUE TITLE
Export large ES index to MINIO 

### ISSUE THREAD

**Issue by grzegorznowacki** [elasticsearch-dump/elasticsearch-dump#788]
I have very large index in Elasticsearch, which is approximately 1TB in size. I want to export it do Minio using compression (`--s3Compress` flag). 

Because the size of the index I am afraid that the operation might cause problems with RAM shortage or disk space shortage. Therefore I would like to ask how exporting index from ES to Minio is internally done. 

Let's assume that:
- I run elasticdump command on my desktop
- I have remote ES cluster and remote Minio

The questions are:
- Is RAM or disk space used on my desktop? Or is the operation performed directly between ES and Minio?
- If it goes directly to Minio - I assume that the index dump does not need to be stored fully in RAM before going to disk and it is done by chunks. But how is the compression made? Does it not need to work on the whole dump file? Or is the compression done on-the-fly?

In a nutshell - do I need to bother about RAM or disk space on my desktop or Minio server?

Additional question:
- In terms of bandwidth and transfering speed - is my desktop used to transfer data between ES and Minio or does the data go directly between them?

**Comment by ferronrsmith** [comment]
1. Is RAM or disk space used on my desktop? Or is the operation performed directly between ES and Minio?
    > RAM will be used to old data in memory (buffering), not so much disk. The operations is directly between ES & minio
2. If it goes directly to Minio - I assume that the index dump does not need to be stored fully in RAM before going to disk, and it is done by chunks. But how is the compression made? Does it not need to work on the whole dump file? Or is the compression done on-the-fly?
   > GZIP compression supports streaming, so as the chunks are being passed they are compressed. I would recommned using the `--fileSize` for faster compression/decompression

**Comment by grzegorznowacki** [comment]
I don't understand that in bullet 1 you say `The operations is directly between ES & minio` and in 3 you say `It's not a direct connection. es-dump (your pc) acts as the middle mand`. 

Does the actual data, I mean `jsonl` go through my PC or directly between ES and Minio?

**Comment by ferronrsmith** [comment]
Yea I edited my answer point 1. The actual data that is pulled from ES.

![Untitled Diagram](https://user-images.githubusercontent.com/159764/115574247-76145880-a28f-11eb-9be5-648df252da38.png)
